File: worker-service/internal/models/job.go
package models

import "time"

type JobStatus string

const (
	StatusPending    JobStatus = "PENDING"
	StatusProcessing JobStatus = "PROCESSING"
	StatusCompleted  JobStatus = "COMPLETED"
	StatusFailed     JobStatus = "FAILED"
	StatusDead       JobStatus = "DEAD"
)

type Job struct {
	ID          string
	Type        string
	Payload     []byte // raw JSON — handlers will unmarshal what they need
	Status      JobStatus
	RetryCount  int
	MaxRetries  int
	Priority    int
	CreatedAt   time.Time
	UpdatedAt   time.Time
	ProcessedAt *time.Time
	ErrorMsg    *string
}

File: api-service/src/types/job.ts
export type JobStatus =
  | "PENDING"
  | "PROCESSING"
  | "COMPLETED"
  | "FAILED"
  | "DEAD";

export type JobType =
  | "send_email"
  | "send_notification"
  | "generate_report"
  | "payment_retry";

export interface Job {
  id: string;
  type: JobType;
  payload: Record<string, unknown>;
  status: JobStatus;
  retry_count: number;
  max_retries: number;
  created_at: Date;
  updated_at: Date;
  processed_at: Date | null;
  error_msg: string | null;
}

export interface CreateJobDTO {
  type: JobType;
  payload: Record<string, unknown>;
  max_retries?: number;
}
